src/Verify.MSTest/Verifier.cs

```diff
@@ -0,0 +1,215 @@
+        {
+            throw new($"TestContext is null. {AttributeUsageHelp}");
+        }
+
+        var testName = CurrentTestContext.Value.TestName;
+        if (testName is null)
+        {
+            throw new($"TestContext.TestName is null. {AttributeUsageHelp}");
+        }
+
+        var typeName = CurrentTestContext.Value.FullyQualifiedTestClassName;
+        if (typeName is null)
+        {
+            throw new($"TestContext.FullyQualifiedTestClassName is null. {AttributeUsageHelp}");
+        }
+
+        if(!TryGetTypeFromTestContext(typeName, CurrentTestContext.Value, out var type))
+        {
+            type = FindType(typeName);
+        }
+
+        var testNameSpan = testName.AsSpan();
+        var indexOf = testNameSpan.IndexOf('(');
+        if (indexOf > 0)
+        {
+            testNameSpan = testNameSpan[..indexOf];
+        }
+
+        indexOf = testNameSpan.IndexOf('.');
+        if (indexOf > 0)
+        {
+            testNameSpan = testNameSpan[(indexOf + 1)..];
+        }
+
+        VerifierSettings.AssignTargetAssembly(type.Assembly);
+        var method = FindMethod(type, testNameSpan);
+
+        var pathInfo = GetPathInfo(sourceFile, type, method);
+        return new(
+            sourceFile,
+            settings,
+            type.NameWithParent(),
+            method.Name,
+            method.ParameterNames(),
+            pathInfo);
+    }
+
+    /// <summary>
+    /// As an optimization, try to retrieve the <see cref="ITestMethod"/> stored on the
+    /// <see cref="TestContext"/>, and use that to retrieve the <see cref="Type"/>.
+    ///
+    /// If reflection fails, return <c>false</c>.
+    /// </summary>
+    /// <param name="typeName">The fully qualified name of the class of the currently running test.</param>
+    /// <param name="testContext">The <see cref="TestContext"/> of the current test.</param>
+    /// <param name="type">The <see cref="Type"/> of the currently running test.</param>
+    /// <returns><c>true</c> if the reflection succeeded; <c>false</c> otherwise.</returns>
+    static bool TryGetTypeFromTestContext(string typeName, TestContext testContext, [NotNullWhen(true)] out Type? type)
+    {
+        try
+        {
+            // We can't use UnsafeAccessor in this context because _testMethod is of type
+            // Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.ObjectModel.TestMethod, which isn't part of the MSTest.TestFramework
+            // package. If / when an API like https://github.com/dotnet/runtime/issues/90081 lands this generic reflection can be replaced.
+            var testMethod = testContext
+                .GetType()
+                .GetField("_testMethod", BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic)
+                ?.GetValue(testContext);
+            var assemblyPath = testMethod
+                ?.GetType()
+                ?.GetProperty("AssemblyName", BindingFlags.Instance | BindingFlags.Public)
```

Comment:
can we cache this property

Reply:
Same rationale as caching the field. We can't speak the `testMethod` type, hence the need for the `.GetType()` call, and the RuntimeTypeCache is caching the `AssemblyName` property lookup for us.

Let me know if I'm misunderstanding the request. Thanks!